File: ruby/Interpreter/expression_reader.rb
require_relative 'number_expression'
require_relative 'add_expression'
require_relative 'subtract_expression'
require_relative 'expression'

class ExpressionReader
  def readToken(tokenList)
    return readNextToken(tokenList)
  end

  def readNextToken(tokenList)
    firstComponentExpression = NumberExpression.new(tokenList[0])
    secondComponentExpression = NumberExpression.new(tokenList[2])

    if (tokenList[1] == '+')
        return __________________ # <<3
    else
        return __________________ # <<5
    end
  end
end

File: java/Interpreter/src/ExpressionReader.java
package exercise;
import java.util.List;

public class ExpressionReader
{
    public Expression ReadToken(List<String> tokenList)
    {
        return ReadNextToken(tokenList);
    }

    private Expression ReadNextToken(List<String> tokenList)
    {
    	NumberExpression firstComponentExpression = new NumberExpression(tokenList.get(0));
    	NumberExpression secondComponentExpression = new NumberExpression(tokenList.get(2));
    	
    	if (tokenList.get(1).contains("+")){
    		return new AddExpression(firstComponentExpression, secondComponentExpression);
    	}else
    		return new SubtractExpression(firstComponentExpression, secondComponentExpression);	
    }
}